lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,534 @@
+    testExecution
+  );
+  t.context.testExecution = testExecution;
+  t.context.executionCumulusId = executionResponse[0];
+
+  t.context.testGranule = generateTestGranule(testCollection, testExecution.arn);
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex(tableNames.files).del();
+  await t.context.knex(tableNames.granules).del();
+  await t.context.knex(tableNames.collections).del();
+  await t.context.knex(tableNames.executions).del();
+});
+
+test.after.always(async (t) => {
+  await granulesModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await executionsModel.deleteTable();
+
+  await s3Utils.recursivelyDeleteS3Bucket(bucket);
+
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
+test.serial('migrateGranuleRecord correctly migrates granule record', async (t) => {
+  const {
+    collectionCumulusId,
+    executionCumulusId,
+    granulePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  const record = await granulePgModel.get(knex, {
+    granule_id: testGranule.granuleId,
+    collection_cumulus_id: collectionCumulusId,
+  });
+
+  t.deepEqual(
+    omit(record, ['cumulus_id']),
+    {
+      granule_id: testGranule.granuleId,
+      status: testGranule.status,
+      collection_cumulus_id: collectionCumulusId,
+      published: testGranule.published,
+      duration: testGranule.duration,
+      time_to_archive: testGranule.timeToArchive,
+      time_to_process: testGranule.timeToPreprocess,
+      product_volume: testGranule.productVolume.toString(),
+      error: testGranule.error,
+      cmr_link: testGranule.cmrLink,
+      execution_cumulus_id: executionCumulusId,
+      pdr_cumulus_id: null,
+      provider_cumulus_id: null,
+      query_fields: null,
+      beginning_date_time: new Date(testGranule.beginningDateTime),
+      ending_date_time: new Date(testGranule.endingDateTime),
+      last_update_date_time: new Date(testGranule.lastUpdateDateTime),
+      processing_end_date_time: new Date(testGranule.processingEndDateTime),
+      processing_start_date_time: new Date(testGranule.processingStartDateTime),
+      production_date_time: new Date(testGranule.productionDateTime),
+      timestamp: new Date(testGranule.timestamp),
+      created_at: new Date(testGranule.createdAt),
+      updated_at: new Date(testGranule.updatedAt),
+    }
+  );
+});
+
+test.serial('migrateFileRecord correctly migrates file record', async (t) => {
+  const {
+    filePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testFile = testGranule.files[0];
+  const granuleCumulusId = await migrateGranuleRecord(testGranule, knex);
+  await migrateFileRecord(testFile, granuleCumulusId, knex);
+
+  const record = await filePgModel.get(knex, { bucket: testFile.bucket, key: testFile.key });
+
+  t.deepEqual(
+    omit(record, fileOmitList),
+    {
+      bucket: testFile.bucket ? testFile.bucket : null,
```

Comment:
Do we not have confidence that bucket/key are set from the test setup? Do we need these ternary expressions?

Reply:
Negatory

Reply:
[`299d0fb` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/299d0fbddc73e17cd00e360974bd4425347f10f3)